api/v1beta1/machine_webhook.go

```diff
@@ -58,7 +58,10 @@ func (m *Machine) Default() {
 	}
 
 	if m.Spec.InfrastructureRef.Namespace == "" {
-		m.Spec.InfrastructureRef.Namespace = m.Namespace
+		// Don't autofill namespace for MachinePool Machines since the infraRef will be populated by the MachinePool controller.
```

Comment:
I'm not sure if we need this, because machine pool machines in this PR are always created with an infrastructure ref